packages/core/src/Cardano/util/primitives.ts

```diff
@@ -38,6 +38,15 @@ export const assertIsBech32WithPrefix = (
   }
 };
 
+/**
+ * Converts a hex string into a typed bech32 encoded string.
+ *
+ * @param prefix The prefix of the bech32 string.
+ * @param hexString The hex string to be encoded.
+ */
+export const toTypedBech32 = <T>(prefix: string, hexString: string): T =>
```

Comment:
We have a type for hex strings, it validates the string.
```suggestion
HexBlob.toTypedBech32 = <T>(prefix: string): T =>
```
You might also want to add
```ts
HexBlob.fromEd25519KeyHash = (hash: Ed25519KeyHash) => hash as unknown as HexBlob;
```

Reply:
> someHexBlob.toTypedBech32('prefix');

You can't really do that without adding it to `String` prototype and I'm not sure we want to do that because it's quite invasive. I suggest to just add it as a "static" fn to be used like `HexBlob.toTypedBech32(someHexBlob)`